### PR TITLE
feat: OPTIC-2: Save draft on leave

### PR DESCRIPTION
**Pull request by robot-ci-heartex**
Hi @Travis1282!

This PR was created in response to a PRs in upstream repo:
- https://github.com/HumanSignal/dm2/pull/231
- https://github.com/HumanSignal/label-studio-frontend/pull/1552

### CODE REVIEW

File: label_studio/frontend/src/components/DraftGuard/DraftGuard.js [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+
+export const DraftGuard = () => {
+  const [currentLocation, setCurrentLocation] = useState(null);
+  const [nextLocation, setNextLocation] = useState(null);
+  const toast = useContext(ToastContext);
+  const history = useHistory();
+
+  useEffect(async () => {
+    if (nextLocation, isFF(FF_OPTIC_7)) {
```
Comment (bmartel) [inline label_studio/frontend/src/components/DraftGuard/DraftGuard.js]:
```suggestion
    if (nextLocation && isFF(FF_OPTIC_7)) {
```

**Review by bmartel — APPROVED** [review]

**Comment by robot-ci-heartex** [comment]
To enable Auto Merge for this PR also merge those PRs:
- https://github.com/HumanSignal/dm2/pull/231

**Comment by bmartel** [comment]
/frontend help
> Command | Description
> --- | ---
> /frontend build           | Get build frontend static for all repos
> /frontend build lsf [sha] | Get build frontend static for HumanSignal/label-studio-frontend only
> /frontend build dm [sha]  | Get build frontend static for HumanSignal/dm2 only

**Comment by bmartel** [comment]
Closing in favor of #4815

**Pull request closed without merging**